### PR TITLE
Update history and change list

### PR DESCRIPTION
**Pull request by minhtuev**
The first draft of the history and change list for MindMeld 4.2

### CODE REVIEW

File: source/integrations/converters.rst [hunk 1]
```diff
@@ -0,0 +1,33 @@
+Mindmeld Conversion Tool
+========================
+
+Introduction
+------------
+
+The MindMeld conversion tool is designed to help make the migration from other conversational AI platforms to MindMeld seamless.
+Currently, we offer support conversions for Rasa and Dialogflow projects.
```
Comment (serapio) [inline source/integrations/converters.rst]:
```suggestion
Currently, we offer conversions for Rasa and Dialogflow projects.
```

**Comment by vembar** [comment]
LGTM

File: source/integrations/converters.rst [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+How does this tool work?
+------------------------
+
+This tool will generate a basic scaffold for a MindMeld project from a given Rasa/DialogFlow project, and will use the
```
Comment (serapio) [inline source/integrations/converters.rst]:
```suggestion
This tool will generate a basic scaffold for a MindMeld project from a given Rasa/Dialogflow project, and will use the
```

File: source/integrations/converters.rst [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+-----
+
+For Rasa project conversion, use the following command:
+
+.. code:: console
+
+    mindmeld convert --rs from_rasa_project_location to_mindmeld_project_location
+
+
+For DialogFlow project conversion, use the following command:
```
Comment (serapio) [inline source/integrations/converters.rst]:
```suggestion
For Dialogflow project conversion, use the following command:
```

File: source/userguide/kb.rst [hunk 4]
```diff
@@ -716,6 +716,8 @@ In the example below, we search for restaurants whose names best match ``firetra
    We can set the ``size`` parameter of the :meth:`execute()` method to specify the maximum number of records.
 
 
+.. _unstructed_data:
```
Comment (serapio) [inline source/userguide/kb.rst]:
```suggestion
.. _unstructured_data:
```

File: source/versions/changes.rst [hunk 5]
```diff
@@ -1,6 +1,96 @@
 Recent Changes
 ==============
 
+
+MindMeld 4.2
+-------------
+
+MindMeld 4.2 packages several new features to allow developers to build non-english NLP applications, do unstructured Q/A searches,
```
Comment (serapio) [inline source/versions/changes.rst]:
```suggestion
MindMeld 4.2 packages several new features to allow developers to build non-English NLP applications, do unstructured Q/A searches,
```

File: source/versions/changes.rst [hunk 6]
```diff
@@ -1,6 +1,96 @@
...
+ and convert RASA and DialogFlow conversation applications to MindMeld applications for ease of on-boarding onto the platform.
```
Comment (serapio) [inline source/versions/changes.rst]:
```suggestion
 and convert Rasa and Dialogflow conversational applications to MindMeld applications for ease of on-boarding onto the platform.
```

File: source/versions/changes.rst (hunk 5 above)
Comment (serapio) [reply source/versions/changes.rst]: Also, QA or Q/A? We've used "QA" elsewhere, but "Q/A" is less ambiguous.

File: source/versions/changes.rst [hunk 7]
```diff
@@ -1,6 +1,96 @@
...
+
+**5. Webex Teams Bot Integration**
+
+MindMeld 4.2 includes built-in support for Webex Teams integration, so developers can seamlessly integrate MindMeld bots
+to Webex Teams. See :doc:`Webex bot integration <../integrations/webex_teams>` for implementation details.
+
+**6. Locale and Language codes**
+
+Locale codes representing the ISO 639-1 language code and ISO3166 alpha 2 country code separated by an underscore
+character, for example, `en_US` and ISO 639-2 language codes, for example, `EN`, can now be used in MindMeld 4.2 to
```
Comment (serapio) [inline source/versions/changes.rst]:
```suggestion
character, such as `en_US` and ISO 639-2 language codes such as `EN`, can now be used in MindMeld 4.2 to
```

File: source/versions/changes.rst [hunk 8]
```diff
@@ -1,6 +1,96 @@
...
+perform NLP tasks on non-english languages.
```
Comment (serapio) [inline source/versions/changes.rst]:
```suggestion
to tailor NLP tasks for the specified locales or languages.
```

File: source/versions/changes.rst [hunk 9]
```diff
@@ -1,6 +1,96 @@
...
+
+Previously, the system entity recognizer, which is rule-based and not trained on app-specific data, could not
+extract system_entities like the time entity ``8`` in the spanish query ``despiértame a las 8``. By passing in the locale
```
Comment (serapio) [inline source/versions/changes.rst]:
```suggestion
extract system_entities like the time entity ``8`` in the Spanish query ``despiértame a las 8``. By passing in the locale
```

File: source/versions/changes.rst [hunk 10]
```diff
@@ -1,6 +1,96 @@
...
+or language code, MindMeld will infer the correct time entity in the above query. We provide a code snippet of how
+to use this functionality below, assuming you have spanish training data for such queries:
```
Comment (serapio) [inline source/versions/changes.rst]:
```suggestion
to use this functionality below, assuming you have Spanish training data for such queries:
```

File: source/versions/changes.rst [hunk 11]
```diff
@@ -1,6 +1,96 @@
...
+                         'value': [ { 'grain': 'hour',
+                                      'value': '2019-02-16T08:00:00.000-08:00'}]}],
+         'intent': 'set_alarm',
+         'text': 'despiértame a las 8'
+       }
+
+
+**7. Stemming**
+
+MindMeld 4.2 allows developers to specify their own custom stemmers, which are useful for non-english languages. We also provide
```
Comment (serapio) [inline source/versions/changes.rst]:
```suggestion
MindMeld 4.2 allows developers to specify their own custom stemmers, which are useful for some languages. We also provide
```

File: source/versions/changes.rst [hunk 12]
```diff
@@ -1,6 +1,96 @@
...
+two in-built stemmers that are wrappers around certain NLTK stemming operations, an English stemmer,
+``mindmeld.stemmers.EnglishhNLTKStemmer`` and a Spanish stemmer, ``mindmeld.stemmers.SpanishNLTKStemmer``. Below, we provide a code
```
Comment (serapio) [inline source/versions/changes.rst]:
```suggestion
``mindmeld.stemmers.EnglishNLTKStemmer`` and a Spanish stemmer ``mindmeld.stemmers.SpanishNLTKStemmer``. Below, we provide a code
```

File: source/versions/changes.rst [hunk 13]
```diff
@@ -1,6 +1,96 @@
...
+                                      'value': '2019-02-16T08:00:00.000-08:00'}]}],
+         'intent': 'set_alarm',
+         'text': 'despiértame a las 8'
+       }
+
+
+**7. Stemming**
+
+MindMeld 4.2 allows developers to specify their own custom stemmers, which are useful for non-english languages. We also provide
+two in-built stemmers that are wrappers around certain NLTK stemming operations, an English stemmer,
```
Comment (serapio) [inline source/versions/changes.rst]:
```suggestion
two built-in stemmers that are wrappers around certain NLTK stemming operations: an English stemmer
```

File: source/versions/changes.rst [hunk 14]
```diff
@@ -1,6 +1,96 @@
...
+snippet on how to use the Spanish built-in stemmer with the NaturalLanguageProcessor object.
+
+.. code:: python
+
+        query_factory = QueryFactory.create_query_factory(app_path, stemmer=SpanishNLTKStemmer())
+        resource_loader = ResourceLoader.create_resource_loader(app_path, query_factory=query_factory)
+        nlp = NaturalLanguageProcessor(app_path, resource_loader)
+
+
+**8. Dialogflow.reprocess**
```
Comment (serapio) [inline source/versions/changes.rst]:
```suggestion
**8. DialogueFlow.reprocess**
```

File: source/versions/changes.rst [hunk 15]
```diff
@@ -1,6 +1,96 @@
...
+
+MindMeld 4.2 includes an improvement to DialogueFlow (a MindMeld dialogue feature) where the user can exit the current dialogue flow and
+return to a default flow. Refer to :ref:`Existing Dialogue Flow <exiting_dialogue_flow>` section on how to exit an existing dialogue flow.
```
Comment (serapio) [inline source/versions/changes.rst]:
```suggestion
return to a default flow. Refer to :ref:`Exiting Dialogue Flow <exiting_dialogue_flow>` section on how to exit an active dialogue flow.
```

File: source/versions/history.rst [hunk 16]
```diff
@@ -1,6 +1,30 @@
...
 
+4.2.0 (2019-09-01)
+------------------
+
+Major Features and Improvements
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+- MindMeld UI is a sample web-based chat client interface to interact with any MindMeld application
+
+- A built-in Question-Answering (QA) component for un-structured data using Elasticsearch
```
Comment (serapio) [inline source/versions/history.rst]:
```suggestion
- A built-in Question-Answering (QA) component for unstructured data using Elasticsearch
```

**Pull request merged**